graphql-language/src/parser.rs

```diff
@@ -0,0 +1,363 @@
+#![allow(dead_code, unused_variables)]
+use ast::*;
+use lexer::Punctuator::*;
+use lexer::Token::{
+  Eof,
+  Punctuator,
+  Name as TokenName,
+  IntValue as TokenInt,
+  FloatValue as TokenFloat,
+  StringValue as TokenString,
+};
+use lexer::{Lexer, Token};
+use source::Source;
+use std::iter::Peekable;
+
+pub type Res<'a, T> = Result<T, &'a str>;
```

Comment:
Just call it result. Also, I'm *pretty* sure your error can be a static reference. Try these types:

```rust
pub type Result<T> = Result<T, &'static str>;
```

`DRes` has a weird kind of code smell for me, and I can't seem to figure it out from browsing the code, what does it do?